src/codewhisperer/service/securityIssueHoverProvider.ts

```diff
@@ -41,6 +41,44 @@ export class SecurityIssueHoverProvider implements vscode.HoverProvider {
         return new vscode.Hover(contents)
     }
 
+    /**
+     * Updates the position of each hover when the text document is changed. The underlying security issues
+     * will be updated with new line numbers to reflect the new positions of the hover.
+     *
+     * @param event Event that triggered the text document change
+     */
+    public updateRanges(event: vscode.TextDocumentChangeEvent) {
+        const changedRange = event.contentChanges[0].range
+        const changedText = event.contentChanges[0].text
+        const lineOffset = this._getLineOffset(changedRange, changedText)
+
+        this._issues = this._issues.map(issues => this._applyRangeOffset(event.document.fileName, issues, lineOffset))
+    }
+
+    private _getLineOffset(range: vscode.Range, text: string) {
```

Comment:
What if the user deleted lines of code where an issue was showing previously? The offset will push the hover content on wrong line numbers. We should remove that issue. 

possible scenarios - 
1. If user removes all the lines where a issue was highlighted.
2. If user partially deletes the lines where a issue is present
3. If user deletes lines such that two issues are getting merged to each other.

Reply:
Good catch, updating it